src/Fantomas.Core.Tests/ConstructorTests.fs

```diff
@@ -0,0 +1,168 @@
+            ) }
+
+    new(info: SerializationInfo, context: StreamingContext) = { inherit Exception(info, context) }
+"""
+
+[<Test>]
+let ``single multiline base constructor, 2335`` () =
+    formatSourceString
+        """
+type FieldNotFoundException<'T>(obj:'T, field:string, specLink:string) =
+    inherit SwaggerSchemaParseException(
+        sprintf "Object MUST contain field `%s` (See %s for more details).\nObject:%A"
+            field specLink obj)
+"""
+        { config with
+            SpaceBeforeClassConstructor = true
+            MaxLineLength = 90 }
+    |> prepend newline
+    |> should
+        equal
+        """
+type FieldNotFoundException<'T> (obj: 'T, field: string, specLink: string) =
+    inherit
+        SwaggerSchemaParseException (
+            sprintf
+                "Object MUST contain field `%s` (See %s for more details).\nObject:%A"
+                field
+                specLink
+                obj
+        )
+"""
+
+[<Test>]
+let ``multiline secondary constructor, 3037`` () =
+    formatSourceString
+        """
+type IntersectionOptions
+    private
+    (
+        primary: bool,
+        ?root: Element,
+        ?rootMargin: string,
+        ?threshold: ResizeArray<float>,
+        ?triggerOnce: bool
+    )
+    =
+
+    new(?root: Element,
+        ?rootMargin: string,
+        ?threshold: ResizeArray<float>,
+        ?triggerOnce: bool) =
+
+        IntersectionOptions(true)
+"""
+        { config with MaxLineLength = 80 }
+    |> prepend newline
+    |> should
+        equal
+        """
+type IntersectionOptions
+    private
+    (
+        primary: bool,
+        ?root: Element,
+        ?rootMargin: string,
+        ?threshold: ResizeArray<float>,
+        ?triggerOnce: bool
+    ) =
+
+    new
+        (
+            ?root: Element,
+            ?rootMargin: string,
+            ?threshold: ResizeArray<float>,
+            ?triggerOnce: bool
+        ) =
+
+        IntersectionOptions(true)
+"""
+
+[<Test>]
+let ``secondary constructor with xml doc`` () =
+    formatSourceString
+        """
+type IntersectionOptions
+    (
+        primary: bool
+    ) =
+
+    /// Good stuff
+    new (secondary: int) = IntersectionOptions(secondary = 0)
+"""
+        config
+    |> prepend newline
+    |> should
+        equal
+        """
+type IntersectionOptions(primary: bool) =
+
+    /// Good stuff
+    new(secondary: int) = IntersectionOptions(secondary = 0)
+"""
+
+[<Test>]
+let ``setting AlternativeLongMemberDefinitions should be respected in long secondary constructor`` () =
+    formatSourceString
+        """
+type StateMachine(
+    // meh
+) =
+    new(
+        // also meh but with an int
+        x:int) as secondCtor = StateMachine()
+"""
+        { config with
+            AlternativeLongMemberDefinitions = true }
+    |> prepend newline
+    |> should
+        equal
+        """
+type StateMachine
+    (
+    // meh
+    )
+    =
+    new
+        (
+            // also meh but with an int
```

Comment:
Hmm, getting some doubts about this one.
In a let binding we only start applying this style today when the paren contains a tuple.
See https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMBDMAFADoBORAdhFVQPQ17oDOsYAtogBZgBGArjgHcAlli65KQ8llIVq1AB4pJWAJRgAvIRUzyOzDm6Edc2vVxMW7Ln0EixEqcZNhFygDRgAnms0EVkN3AoRmwsSQBzRlQwAG0giFATSAASLE8AB2Q0SGUA%2BKoUgDdzXizY-OoABkDnKABJciQpAGUhAC9kGucAFgqwAF18gF8uhL6UtMzonKk8pOTi9FLouNqARlG5SABZXHkAGUlEfcRycNE55zWAJkq%2BweoR-MSt1IyyyFP4AHkAM0PyIhmmlMJdqEUSmVVs5rptwQBRRp-AGdPoQSDoX6Qe7DTYvcFvKbZbiwWDoMEFBaQlZosAAZjhBQaIRIWAAYpJzAA5RACdBHCnULAkUo4x548aEj4ksmCqBUpZQ2ndRno5rpXDQRAAIUQv1gJEQAAVcCRcOwsIgSHKIMLRSYHlQntR8ZTJtLSeTGRDFTTagBWVVQdWanV6g3HWACK3QXAhBqFWCxsKwcg2sB2xBip0S%2Bbu6Yyr2SxbLNDQkwANiDkBDWt1%2BsNAFV0pkSLH4%2BRE8mhKn079zCFsxBnVRXeipQXPXKfaXyrUAOzV2thhuIADC6DjjDXqcYdugWANfYHWYduOekvzxKn3oVs-LcgAHEuNXXw4btohWNwrcemKe5EdYdc1eK8oELac7yVWoAE4X1DesIx3dBeyDft-yHMARzGPN3knWVbxLaCrmqWka1fRAAEFfktEgd1YVhcHTTNMOwsAx3lMDIAgwjqTLWk1g2Mjl0Qw1mi-IQ4BQtM0JPViQIJLieOLPi5yuWFhIo6jaPE1hJLJVDaRYs9xQvXCiXAm8VN9fj1gZTTQ0okhYF4RoABFEH5PTaOYkUAOoICsIUt08OyXJeJstSTDWFUyN2eQ6l%2BAAVDhTmaDgDSwAB1IR4AuIM7hMnMzNA0KoHC6z7wEwM4r2RKUtOeF-3SzKcryjh0wrQrAPPF1LzKmYsEgoi-SuKtaoS8hfiEeRvlbXBDxIeF5HSQ1GEYHsZNpR9uoC3rR36izBuG1SH2oNZFwmgAlRA4BIeA2vy5VdqoQK2I4iYBoq%2BYRtsq5n2u26DXgLleG-K0-jqS1WCiIM1nkkrFIG1heHQMJ%2BUBNkDUYrBaKSvCIqq9Y4LIm67vgbZUfRo4sZIHGfOraAOFNTVaIAfWEdrsSK4DEZCo7vteX6orkW5q3ipyzU8b4SEOPdHo6oMdoRvrzI%2BQWCWFs6qGuITah2PZJdwaXZaEPdQfBkhIeh2GBJVg61emFG0aEDHEFp%2BmrXxqZCeIkxrg0-WjZNuWsEpl23Y9haGbIpmWYPK0Odyi57Zw0qBdmX3Rv9%2Bz9figA1SFtUkeAIgV9NlZ5oK%2BfHLiNcpLXaWuWK872NlXIPTbi8aMvk8V57U-Yw71czyq-dFmrW-kNzYCwABxbBltWxB1s28ulZeiA3uC2uBuUn7Tqb8b9Z5Pkjl1LBo1Ob3EA86byBETbKMaT9Ldt2pjJ60zVfTj0CLH7OotLpB35OEcg7dyCd1TK0MBC1eCGiSrABoTQsALU2n%2BQcVd3rD3wkWA%2BkVtYQGuADEBtFyBoMKJGM4r8fwkDvpIR%2Bu4MH%2BVevtNOSMjr7yFofWo1wSZ5ypkIFE%2BxzTcHgLgDcsANpnFPm7Zhg8PoTmvP-fBRNnB0lIvrJeVohDsCkOYAA0ogRA6RkGnCwA0bUZpIH91qOhTBX9io-w4X-PB3CCG0jpHrZwkBtSbnIAAaxRIwJyLlGg8j3IgCmgi3ZLzWhtJhQZP57W-g7X%2BuCToeNqHSQOPjtSmlErfM20ASC6M5JaeAjYH6pg8tATcZoUxbTsXJLBO9OLIxiefM00AAnYGBJ4UEWc-omDpLnHx4dqaAisZqXpWB%2BmDNjmaVgmR4Dc0cbzZx-MR5DSGSLagdIW4%2BKMSY%2BKFtaF-D8eIIJRx34kU3gMNhQ9HbKLcZrHh6jJ4%2BNkefd8iAJmuyOPRdI-A0GpjiSvBJTTnDJNYak9hWyMm7MIfSY%2BPjtGlL0ag9ATU9KMFsc4exLCt7DCCP0LoORGBsg2tEQlIAhhAA

Reply:
Yeah, this should only apply for tuples. [The guide](https://learn.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-function-and-member-arguments) does mention this.